trlx/trainer/accelerate_base_trainer.py

```diff
@@ -196,15 +196,13 @@ def generate_eval(self, input_ids, attention_mask=None, **kwargs):
                 input_ids=input_ids, attention_mask=attention_mask, **kwargs
             )
 
-    def save(self, directory=None):
-        """Creates checkpoint of optimizer, scheduler and a model"""
-        self.accelerator.save_state(directory or self.config.train.checkpoint_dir)
-        if directory:
-            self.model.base_model.save_pretrained(f"hf_model_{directory}")
-        else:
-            self.model.base_model.save_pretrained(
-                f"hf_model_{self.config.train.checkpoint_dir}"
-            )
+    def save(self, directory: Optional[str] = None):
+        """Creates a checkpoint of the optimizer, scheduler and model"""
+        directory = directory or self.config.train.checkpoint_dir
+        self.accelerator.save_state(directory)
+        self.accelerator.unwrap_model(self.model).base_model.save_pretrained(
+            f"{directory}/hf_model"
+        )
```

Comment:
How about adding a seperate method `save_pretrained` to the `trainer` instead, delegated to each model since for ILQL saving `base_model` is not enough and it has to be wrapper over `PreTrainedModel` to achieve excepted result

Reply:
Agreed. @jon-tow make sure to update the readme as well.

Reply:
@reciprocated If you wrap the ILQL model with `transformers.PreTrainedModel` (or inherit it) and run `save_pretrained`, you still don't get the custom `generate` method needed for inference. I'm not sure what to do here because there isn't a reason to have `save_pretrained` if the custom generation implementation is not compatible with HF loading. Any hints/suggestions?

Reply:
ILQL is kinda like GeDi in that regard. Doesn't GeDi already exist on the hub?

Reply:
This is what Aman had suggested I believe, ILQL model will inevitably be tied to this repo. We can do the changes necessary at a later point, maybe `save_pretrained` on the base trainer with a `raise NotImplementedError` would be good for now

Reply:
Done. @reciprocated thanks for the suggestions 👍👍